iothub_client/tests/common_e2e/iothubclient_common_e2e.c

```diff
@@ -52,10 +53,11 @@ static TEST_PROTOCOL_TYPE test_protocol_type;
 const char* TEST_EVENT_DATA_FMT = "{\"data\":\"%.24s\",\"id\":\"%d\"}";
 const char* TEST_EVENT_DATA_FMT_SPECIAL_CHAR = "{\"#data\":\"$%.24s\",\";id\":\"*%d\"}";
 
-const char* MSG_ID = "MessageIdForE2E";
-const char* MSG_ID_SPECIAL = "MessageIdForE2E*";
-const char* MSG_CORRELATION_ID = "MessageCorrelationIdForE2E";
-const char* MSG_CORRELATION_ID_SPECIAL = "MessageCorrelationIdFor#E2E";
+#define MSG_UNIQUE_ID_STAMP_LEN 5
+#define MSG_ID "MessageIdForE2E"
+#define MSG_ID_SPECIAL "MessageIdForE2E*"
```

Comment:
Nit - could we keep the `const char` stuff so we're less pre-processor heavy?

You can below to make the `sizeof` work.
```c
const char* MSG_ID[] = "..."
```